docs/data_driven_testing.adoc

```diff
@@ -252,6 +252,11 @@ used as a data provider. This includes objects of type `Collection`, `String`, `
 they can fetch data from external sources like text files, databases and spreadsheets, or generate data randomly.
 Data providers are queried for their next value only when needed (before the next iteration).
 
+NOTE: For performance reasons it is helpful to use `Iterable.iterator()` instead of the `Iterable` as data pipe,
+      if a custom `size()` method is not available or the calculation of the `Iterable` is expensive.
+      The Groovy `Iterable` contract contains a `size()` method, which iterates the `Iterable` to calculate the size.
```

Comment:
would specifying that iteration over the Iterable may be done multiple times before execution be a more accurate way of expressing the internal behavior? In my opinion, a reader could assume that the iteration is only done once from reading this. The implication that the next value is fetched unnecessarily also seems to contradict the statement in 253.